standardtracer/span.go

```diff
@@ -11,22 +11,33 @@ import (
 // Implements the `Span` interface. Created via tracerImpl (see
 // `standardtracer.New()`).
 type spanImpl struct {
-	lock     sync.Mutex
-	tracer   *tracerImpl
-	recorder SpanRecorder
-	raw      RawSpan
+	tracer     *tracerImpl
+	sync.Mutex // protects the fields below
```

Comment:
`spanImpl` had a `Mutex` even before, and generally I would only use an `RWMutex` when it's indicated by a specific workload (since it has generally higher overhead). I would naively expect a trace to be "mostly" written to, so that a `Mutex` seems the better fit.